### PR TITLE
#129 Bug/new admin cannot have company company tags users career paths

### PR DESCRIPTION
**Pull request by Pavel1066**

### CODE REVIEW

File: backend/src/api/controllers/user-managment.controller.ts [hunk 1]
```diff
@@ -21,31 +22,47 @@ import { UserMissingDataForm } from '~/common/forms/user.forms';
...
+
+  const company: Company = await companyRepository.findOne({
+    id: user.company.id,
+  });
+
+  const newUser = await registerUser(
     createDefaultUser(email),
     roleType,
-    companyId,
+    company.id,
```
Comment (aleksandr-kuharenko) [inline backend/src/api/controllers/user-managment.controller.ts]: you can use here `user.company.id` as you already checked it is not null. no need to query again

**Review by aleksandr-kuharenko — CHANGES_REQUESTED** [review]

**Review by aleksandr-kuharenko — APPROVED** [review]

File: backend/src/api/controllers/career-path.controller.ts [hunk 2]
```diff
@@ -75,8 +78,16 @@ const getLevelsAndSkills = async (domain: Domain): Promise<DomainLevel[]> => {
 
 export const createDomainTree = async (
   data: CareerPath,
-  company: Company,
+  userId: string,
 ): Promise<CareerPathResponse> => {
+  const userRepository = getCustomRepository(UserRepository);
+  const user = await userRepository.geUserById(userId);
```
Comment (karyna-kh) [inline backend/src/api/controllers/career-path.controller.ts:84]: method name geUserById has a mistake

File: backend/src/services/user.service.ts [hunk 3]
```diff
@@ -165,11 +165,16 @@ export const authenticateUser = async (data: UserLoginForm): Promise<User> => {
...
 ): Promise<IListUser[]> => {
   const userRepository = getCustomRepository(UserRepository);
+  const user = await userRepository.geUserById(userId);
+
+  if (!user.company) {
+    throw badRequestError('User doesn`t create company!!!');
+  }
 
-  const usersList = await userRepository.getUsersByCompamyId(companyId);
+  const usersList = await userRepository.getUsersByCompamyId(user.company.id);
```
Comment (karyna-kh) [inline backend/src/services/user.service.ts:177]: method name getUserByCompamy has a mistake

File: backend/src/api/controllers/career-path.controller.ts [hunk 4]
```diff
@@ -135,9 +146,17 @@ export const createDomainTree = async (
 };
 
 export const getDomainTrees = async (
-  company: Company,
+  userId: string,
 ): Promise<CareerPathResponse[]> => {
   const careeerPathResponse: CareerPathResponse[] = [];
```
Comment (karyna-kh) [inline backend/src/api/controllers/career-path.controller.ts:151]: method name careeerPathResponse has a mistake

**Pull request merged**